extensions/amp-ad-network-doubleclick-impl/0.1/safeframe-host.js

```diff
@@ -219,6 +221,25 @@ export class SafeframeHostApi {
     return attributes;
   }
 
+  /**
+   * Returns the canonical URL of the page, if the publisher allows
+   * it to be passed.
+   * @return {string|boolean}
+   */
+  maybeGetCanonicalUrl() {
+    // Don't allow for referrer policy same-origin,
+    // as Safeframe will always be a different origin.
+    // Don't allow for no-referrer.
+    if (!!this.win_.document.querySelector(
+        "meta[name='referrer'][content='same-origin']") ||
+        !!this.win_.document.querySelector(
```

Comment:
If origin, we should just return the origin of the canonical url (e.g. nytimes.com/foo?blah becomes nytimes.com)

Reply:
done